### ISSUE TITLE
Removing the last separator in the Multiple answer code

### ISSUE THREAD

**Issue by Scoubines** [scott2000/answerset#37]
Hi

I'm using your great addon called "Answer Set - Multiple answers for "type in the answer" cards" and there is a little bug that I'd like to fix.

Currently I've put "," as an alternative separator for my type-in answers but I don't want the very last comma to show up in the answer displayed. So let's say the question is 126, currently the answers displayed are: 7x18, 18x7, 9x14, 14x9,

But I would like to have displayed: 7x18, 18x7, 9x14, 14x9
So basically the same without the final "," at the end. It's purely aesthetic I know but that would be nice.

From Chat GPT it looks like it's around the [compare.py] file but I wasn't able to do much from here.
Could you update it or if it's too cumbersome give me the fix for it?

Thank you so much,

**Comment by scott2000** [comment]
Hi, thanks for the bug report! Yeah, I definitely think it shouldn't be showing the extra trailing comma. I think I know what's causing this, so I'll look into it and see if I can release an update for it sometime this week.

**Comment by Scoubines** [comment]
Awesome! Thank you so much! Can't wait to see the update!

**Comment by scott2000** [comment]
I'm having a bit of trouble replicating this issue actually, so would you be able to send me the add-on configuration from Anki?

**Comment by Scoubines** [comment]
Hi, sure thing. Here it is.
I used it in my case to train on multiplication tables, which are on the format a numberXanother number, followed by a "," when I have other options like 126 in question giving me 7x18, 18x7, 9x14, 14x9,

The complete add on is joined and the configuration is as followed:

{
    "Enable Answer Choice Comments [...]": false,
    "Enable Answer Comments (...)": false,
    "Enable Lenient Validation": true,
    "Equivalent Strings": [],
    "Ignore Case": true,
    "Ignore Separators in Brackets": true,
    "Ignored Characters": "x,",
    "Numeric Comparison Factor": 1.0,
    "Separators": " ,"
}

As you see I put "x" as an ignored character (so that I can just put "187" for more speed type-in, and a space before the",", so that I can see more clearly the separation in the answer, so  7x18, 18x7, 9x14, 14x9, and not  7x18,18x7,9x14,14x9, but I double any of that has an incident on the bug?

Thank you!
[1827331674.zip](https://github.com/user-attachments/files/16608034/1827331674.zip)

**Comment by scott2000** [comment]
So I think what's actually happening is that it always uses the first separator from "Separators" which is present in the correct answer. So in this case, that means that the separator is just a space, since both comma and space are present, but space comes first in `" ,"`. That means it's splitting up the answer by looking only at spaces, so the answers it sees are:

* `7x18,`
* `18x7,`
* `9x14,`
* `14x9,`

That's why the comma is being shown even though normally it shouldn't be, because the comma is being treated as part of the answer itself rather than a separator. If you use `", "` instead of `" ,"` for the separators, maybe that could solve the issue, since that would split based on the comma instead whenever a comma is present. Could you try that out and let me know if it fixes the issue?

**Comment by Scoubines** [comment]
Hi Scott,

Thank you for looking into it!
So I've tried your idea but it didn't work as now I've lost all visual separation on the answer step as you can see.
Sure I can read it but not ideal for the eyes to see clearly the separate operations in my case

My setup is as follow just to be sure:

{
    "Enable Answer Choice Comments [...]": false,
    "Enable Answer Comments (...)": false,
    "Enable Lenient Validation": true,
    "Equivalent Strings": [],
    "Ignore Case": true,
    "Ignore Separators in Brackets": true,
    "Ignored Characters": "x,",
    "Numeric Comparison Factor": 1.0,
    "Separators": ", "
}


I've joined the exact inputs and ouputs so you can have a better maybe?

![Xnip2024-08-15_11-07-51](https://github.com/user-attachments/assets/cf160e46-f610-4f74-8429-37744a220cb8)

**Comment by scott2000** [comment]
Oh that's really strange. I'm not really sure why the commas are disappearing, they're definitely not supposed to. Do you happen to have any custom CSS styling on the card?

**Comment by Scoubines** [comment]
Oh you were right, it does come from my custom type in note type template actually! When I made a basic type in with no edit it did worked perfectly as you see.
![Xnip2024-08-15_15-25-56](https://github.com/user-attachments/assets/6a2154c4-1e42-46d2-866d-ba7c930935e9)

But as I'd like to keep my own custom template, would you kind enough to point out what's wrong on mine causing the bug?

Here is my CSS (supposing it's not a Front/Back custom issue). It's a bit lengthy with for sure weird lines but it works great apart from this issue :

<div style="display:flex">
  <img src="" style="display:block !important; margin:auto">
</div>
/* center #qa inside .card (the body)  */

.card {
  font-size: 1.4rem;
  text-align: center;
  background-color: #8ab0ae;
  --text-fg: #575e6f;
  --text-bg: rgba(255, 255, 255, .5);
  --kard-bg: #fed6bc;

}

/* make #qa (child of .card) centerable */
/* and left-aligned */
#qa {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.kard {
  position: relative;  
  padding: 2em 7rem; /* Reduced padding for better display */
  background: #3D3847;
  box-shadow: rgba(0, 0, 0, 0.30) 0px 0px 25px 4px;
  border-radius: 15px;
  min-height: 240px;
  max-height: 240px;
  min-width: 150px;
  max-width: 80%; /* Adjusted max-width for responsiveness */
  text-align: center;
  color: white;

}


.parafront {  
  position: relative;
  top: 1px;
	left: -50px;
  width: 400px;
  margin: auto; /* Alignement horizontal */
  display: block;
  text-align: left; /* Centrage horizontal du texte */
  color: white;
  text-shadow: 0.1rem 0.1rem 0 var(--text-bg); /* Valeur de l'ombre du texte corrigée */
  height: 10px;

}

input[type=text], #typeans {
  background-color: rgba(0,0,0,0);
 top: 11px;
  left: -16px; /* Utilisation de la même valeur de positionnement que .parafront */ 
  text-align: left;
  cursor: text;
  border-radius: 5px;
  border-style: none; /* Suppression de la bordure */
  border-width: 0; /* Suppression de la bordure */
  margin-top: px;
  font-size: 40px;
  font-family: Arial;
 	font-weight: normal
  position: relative;
	width:300px;

 
}

input { 
border-style: hidden; 
height:38px  !important;  
}

#typeans {....Mon type-in....
  font-family: Arial;
	font-size: 40px;

  position: fixed;
  top: calc(32% + 1rem); /* Positionner l'élément 1 rem sous la ligne */
  left: 105.5%;
  transform: translate(-122%, 0); /* Ajuster la transformation */
  text-align: left;
	width:390px;
}

.extra {
  display: block;
  position: absolute;
  top: 95%;
  left: 46%;
  width: 400%;
  max-width: 350px;
  transform: translate(-50%, -100%);
  margin-top: px; /* Adjust as necessary */
  max-height: 75px; /* Adjust as necessary */
  overflow-y: auto; /* Enable vertical scrolling if needed */
  color: CornflowerBlue;
  font-size: 15px;
  text-align: left
}

.card { /* question */
 font-size: 40px;
	font-family: Futura;
 color: black;
    margin: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
 width: 150px;
  height: 200px;


}

input[type=text]:focus {

}

#answer {
  position: absolute;
  top: 28%;
  left: 49%;
  transform: translate(-50%, -50%);
  width: 385px;
}


.extra::-webkit-scrollbar {
  width: 5px; /* Adjust scrollbar width */
}

.extra::-webkit-scrollbar-track {
  background: #f1f1f1;
}

.extra::-webkit-scrollbar-thumb {
  background-color: #888;
  border-radius: 10px;
}

.extra::-webkit-scrollbar-thumb:hover {
  background-color: #555;
}





#qa .kard {
  position: relative;
}

#qa .kard .parextra {}

#qa .kard .paraback {
  display: none;
}

.backtemplate #qa .kard .paraback {
  display: block;
}

.expression {
  font-weight: bold;
  background: #8fbc8f;
  color: black;
}

/* Hide hint on Front, display on Back */
#hint {
  display: none;
  margin-top: 10px; /* Adjust as necessary */
}
.backtemplate #hint {
  display: block;
}


div #correct{
font-family: "Avenir";,
}

div #incorrect{
font-family: "Avenir";

}


  color: CornflowerBlue;
  font-size: 15px;
}

.extra::-webkit-scrollbar {
  width: 5px; /* Adjust scrollbar width */
}

.extra::-webkit-scrollbar-track {
  background: #f1f1f1;
}

.extra::-webkit-scrollbar-thumb {
  background-color: #888;
  border-radius: 10px;
}

.extra::-webkit-scrollbar-thumb:hover {
  background-color: #555;
}


}


#qa .kard {
  position: relative;
}

#qa .kard .parextra {}

#qa .kard .paraback {
  display: none;
}

.backtemplate #qa .kard .paraback {
  display: block;
}


/* Hide hint on Front, display on Back */
#hint {
  display: none;
  margin-top: 10px; /* Adjust as necessary */
}
.backtemplate #hint {
  display: block;
}


div #correct{
font-family: Arial;,
}

div #incorrect{

font-size: 40px;
font-weight:bold;

}


@media (min-width: 500px) {
  #qa {
    min-width: 500px;
    max-width: 500px;
  }

**Comment by scott2000** [comment]
Hmm the only thing I can think of is that maybe `...Mon type-in....` needs to be in a comment like `/* ...Mon type-in.... */`, otherwise it might mess up the styling, but I'm not sure if that would cause this issue or not.

I'm away from my computer right now, but I'll take a look at it when I get home later and try to figure out what's causing this to happen.

Also, just a note on GitHub formatting: you can format text as CSS code by surrounding it like this:

~~~
```css
YOUR CSS HERE
```
~~~

**Comment by Scoubines** [comment]
Oh right, tried to insert with the CSS code button but thanks for the tip!

So I've just tried to change the comment into /* ...Mon type-in.... */ but with no avail.. Still have the same result.
So sure take all the time when you can have a look. I wish I had more knowledge on code to help you or something but that's getting out my scope T_T

Thank you again!

**Comment by Scoubines** [comment]
Hi again!

I solved it! I tried to erase all my CSS to see and nothing changed so I figured out that it either comes from Front or Back and by trial and errors with ChatGPT I had a final good fix with the Back side that was probably caused by a java script.
Thank you again for your time; I will gladly use your add on a lot now! ;D

**Comment by scott2000** [comment]
Oh that's great! I'm glad you were able to find a solution!

> Thank you again for your time; I will gladly use your add on a lot now! ;D

No problem! I'm glad it's useful :)

Let me know if you run into any more issues!
